tests/NCronJob.Tests/RunAtStartupJobTests.cs

```diff
@@ -0,0 +1,149 @@
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Shouldly;
+
+namespace NCronJob.Tests;
+
+public class RunAtStartupJobTests : JobIntegrationBase
+{
+    [Fact]
+    public async Task UseNCronJobShouldTriggerStartupJobs()
+    {
+        var builder = Host.CreateDefaultBuilder();
+        var storage = new Storage();
+        builder.ConfigureServices(services =>
+        {
+            services.AddNCronJob(s => s.AddJob<SimpleJob>().RunAtStartup());
+            services.AddSingleton(_ => storage);
+        });
+        using var app = builder.Build();
+
+        await app.UseNCronJobAsync();
+
+        storage.Content.Count.ShouldBe(1);
+        storage.Content[0].ShouldBe("SimpleJob");
+    }
+
+    [Fact]
+    public async Task ShouldStartStartupJobsBeforeApplicationIsSpunUp()
+    {
+        var builder = Host.CreateDefaultBuilder();
+        var storage = new Storage();
+        builder.ConfigureServices(services =>
+        {
+            services.AddNCronJob(s => s.AddJob<SimpleJob>().RunAtStartup());
+            services.AddSingleton(_ => storage);
+            services.AddHostedService<StartingService>();
+        });
+        using var app = builder.Build();
+
+        await app.UseNCronJobAsync();
+        await RunApp(app);
+
+        storage.Content.Count.ShouldBe(2);
+        storage.Content[0].ShouldBe("SimpleJob");
+        storage.Content[1].ShouldBe("StartingService");
+    }
+
+    [Fact]
+    public async Task StartupJobThatThrowsShouldNotPreventHostFromStarting()
```

Comment:
Will there be a way in the future to control that behavior?

Reply:
Sure - if someone wishes that. For now, we have the `IExceptionHandler`